service/history/tasks/predicates.go

```diff
@@ -150,6 +150,21 @@ func OrPredicates(a Predicate, b Predicate) Predicate {
 	return predicates.Or(a, b)
 }
 
+func IsUniverisalPredicate(p Predicate) bool {
+	_, ok := p.(*predicates.UniversalImpl[Task])
```

Comment:
UniversalImpl does not really need type parameter.

Reply:
Logically yes. But `UniversalImpl` needs to have the `Test(T) bool` method and golang currently doesn't allow parameterized method.

https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#no-parameterized-methods